provider/azureProvider.js

```diff
@@ -178,7 +179,24 @@ return new BbPromise((resolve, reject) => {
       }
     }
 
-    const template = JSON.parse(fs.readFileSync(templateFilePath, 'utf8'));
+    let template = JSON.parse(fs.readFileSync(templateFilePath, 'utf8'));
+
```

Comment:
See point 2. at https://github.com/serverless/serverless-azure-functions/pull/61#issuecomment-297384088 for the explanation why I did not use parameters.

To elaborate, it would be easy to pass the environment variables as parameters (as you say), but I do not know a syntax in the ARM template JSON to loop through those parameters in the part of the JSON where that would be needed. If you do know this would be possible, please send a diff to the ARM template that shows how the looping of the parameters would be done under the list which is the value of the `appSettings` property.